pkg/activator/handler/handler.go

```diff
@@ -34,30 +33,53 @@ type ActivationHandler struct {
 	Logger    *zap.SugaredLogger
 	Transport http.RoundTripper
 	Reporter  activator.StatsReporter
+	Throttler *activator.Throttler
 }
 
 func (a *ActivationHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	namespace := pkghttp.LastHeaderValue(r.Header, activator.RevisionHeaderNamespace)
 	name := pkghttp.LastHeaderValue(r.Header, activator.RevisionHeaderName)
-
 	start := time.Now()
-	capture := &statusCapture{
-		ResponseWriter: w,
-		statusCode:     http.StatusOK,
-	}
+	revID := activator.RevisionID{namespace, name}
 
+	// ActiveEndpoint() will block until the first endpoint is available.
 	ar := a.Activator.ActiveEndpoint(namespace, name)
+
 	if ar.Error != nil {
 		msg := fmt.Sprintf("Error getting active endpoint: %v", ar.Error)
 		a.Logger.Error(msg)
 		http.Error(w, msg, ar.Status)
 		return
 	}
-
 	target := &url.URL{
 		Scheme: "http",
 		Host:   fmt.Sprintf("%s:%d", ar.Endpoint.FQDN, ar.Endpoint.Port),
 	}
+
+	err := a.Throttler.Try(revID, func() {
+		attempts, httpStatus := a.proxyRequest(w, r, target)
+
+		// Report the metrics
+		duration := time.Since(start)
+
+		a.Reporter.ReportRequestCount(namespace, ar.ServiceName, ar.ConfigurationName, name, httpStatus, attempts, 1.0)
+		a.Reporter.ReportResponseTime(namespace, ar.ServiceName, ar.ConfigurationName, name, httpStatus, duration)
+	})
+	if err != nil {
+		if err == activator.ErrActivatorOverload {
+			http.Error(w, activator.ErrActivatorOverload.Error(), http.StatusServiceUnavailable)
```

Comment:
I'm wondering if this should be using a 429?

Reply:
Generally agree, but I think it would lead to inconsistency with the 503s we send back from the queue-proxy overflow. So I prefer to leave it with 503 for now, we could replace it 429 once we redirect the queue-proxy overflow back to the activator.

Reply:
Can you open an issue to discuss the use of 429 vs. 5xx?

It is sort of cosmetic, but I think 429 is a much clearer status code (at least to me).  503 and 504 seem to grow on trees :)

Reply:
I vaguely recall discussing this somewhere. The bottomline was:

- If your system runs out of resources (as in this case), return 503.
- If the user exhausted his quotas, use 429.

Reply:
Got it, that depends on whether we consider the size of a breaker to be a user limit or not. Technically if the user is running out of throttler tokens, we might still have the capacity to provision containers. 

I opened this issue:
https://github.com/knative/serving/issues/3166